### PR TITLE
Implement a new method for tox 4.18+ 

### PR DESCRIPTION
**Pull request by hroncok**
Fixes https://github.com/fedora-python/tox-current-env/issues/77

### CODE REVIEW

**Comment by hroncok** [comment]
I am afraid our tests are broken due to https://github.com/fedora-python/tox-current-env/issues/74 and https://github.com/fedora-python/tox-current-env/issues/75

But locally when I run _some_ tests, it seems to pass with tox 4.18.0 (and it was a huge fail without this method).

Would appreciate it if you could verify this fix, @kevalmorabia97

**Comment by hroncok** [comment]
This doesn't work when custom base_python is set.

**Comment by kevalmorabia97** [comment]
Verified. This change fixes the issue

**Comment by kevalmorabia97** [comment]
Once this is merged, can you please make a new release?

**Comment by hroncok** [comment]
That is the plan.

File: src/tox_current_env/hooks4.py [hunk 1]
```diff
@@ -209,6 +209,12 @@ def prepend_env_var_path(self):
     def runs_on_platform(self):
         return sys.platform
 
+    @classmethod
+    def python_spec_for_path(cls, path):
+        # https://github.com/fedora-python/tox-current-env/issues/77
+        ver, is_64 = sys.version_info, sys.maxsize != 2**32
```
Comment (frenzymadness) [inline src/tox_current_env/hooks4.py:215]: `sys.maxsize` is `2**31-1` for 32 bit and `2**63-1` for 64bit so the condition should be:

```suggestion
        ver, is_64 = sys.version_info, sys.maxsize > 2**32
```

File: src/tox_current_env/hooks4.py [hunk 2]
```diff
@@ -209,6 +209,12 @@ def prepend_env_var_path(self):
...
+        return PythonSpec.from_string_spec(f"{sys.implementation}{ver.major}{ver.minor}-{is_64}")
```
Comment (frenzymadness) [inline src/tox_current_env/hooks4.py:216]: I'm not sure about the last part. The regex for parsing this is:

```python
PATTERN = re.compile(r"^(?P<impl>[a-zA-Z]+)?(?P<version>[0-9.]+)?(?:-(?P<arch>32|64))?$")
```

So I think that `is_64` should contain 32/64 and not a boolean value.

Comment (hroncok) [reply src/tox_current_env/hooks4.py:216]: I copy-pasted this from tox itself. If it's wrong, so is tox :(

In https://github.com/tox-dev/tox/pull/3325 this was added:

https://github.com/tox-dev/tox/blob/dde496464b25484720ea4327429ae17cafcf1131/src/tox/tox_env/python/api.py#L175-L176

```python
                    if str(spec_base.path) == sys.executable:
                        ver, is_64 = sys.version_info, sys.maxsize != 2**32
                        spec_base = PythonSpec.from_string_spec(f"{sys.implementation}{ver.major}{ver.minor}-{is_64}")
```

**Review by frenzymadness — APPROVED** [review]

**Pull request merged**